# ðŸ¦  My Microbiome toolkit
There are scripts for microbiome analysis, metagenome assembly pipelines, and visualization of results of programs, which are commonly used in metagenome analysis. 

![Logo2_small](https://github.com/Chartiza/Microbiome/assets/15068419/13777084-de09-4425-9fd5-d9be715c548a)

- <b> checkM1_vs_checkM2 </b> - comparison of the performance of programs tested on complete circulated genomes from different phyla <br>
- <b> NCBI_vs_GTDB </b> - comparison of Bacterail & Arhaeal in two databases <br>
- <b> Program_Comparison </b> - comparing bin quality between binners and MAGpurify improving performance <br>
- <b> RAD </b> - exploratory metagenomic analysis scripts <br>
- <b> gtdb_vizualization </b> - Bacteria and Archae taxonomy <br>
- <b> gunc_visualization </b> - GUNC output visualization of contig annotations <br>
- <b> ShortReads_from_LongReads </b> - converting long Pacbio reads into short PE Illumina reads <br>

