# ðŸ¦  My Microbiome toolkit
There are scripts for microbiome analysis, metagenome assembly pipelines, and visualization of results of programs, which are commonly used in metagenome analysis. 

ðŸ¤— Folders: 

<i> checkM1_vs_checkM2 </i> - comparison of the performance of programs tested on complete circulated genomes from different phyla <br>
<i> NCBI_vs_GTDB </i> - comparison of Bacterail & Arhaeal in two databases <br>
<i> Program_Comparison </i> - comparing bin quality between binners and MAGpurify improving performance <br>
<i> RAD </i> - exploratory metagenomic analysis scripts <br>
<i> gtdb_vizualization </i> - Bacteria and Archae taxonomy <br>
<i> gunc_visualization </i> - GUNC output visualization of contig annotations <br>
<i> ShortReads_from_LongReads </i> - converting long Pacbio reads into short PE Illumina reads <br>

<img width="1323" alt="Pic_genus" src="https://user-images.githubusercontent.com/15068419/178560631-8c3c1f76-7d1b-4b04-8a61-7b80e09b7c57.png">

