# ðŸ¦  My Microbiome toolkit
There are scripts for microbiome analysis, metagenome assembly pipelines, and visualization of results of programs, which are commonly used in metagenome analysis. 

![alt text](we-are2_small2.PNG)

- <b>[**checkM1_vs_checkM2**](./checkM1_vs_checkM2) </b> - comparison of the performance of programs tested on complete circulated genomes from different phyla <br>
- <b> [**NCBI_vs_GTDB**](./NCBI_vs_GTDB) </b> - comparison of Bacterail & Arhaeal in two databases <br>
- <b> [**NCBI_tax_id**](./NCBI_tax_id) </b> - assign NCBI tax-id to species from GTDB wich has match in NCBI <br>
- <b> [**Program_Comparison**](./Program_Comparison) </b> - comparing bin quality between binners and MAGpurify improving performance <br>
- <b> [**RAD**](./RAD) </b> - exploratory metagenomic analysis scripts <br>
- <b> [**gtdb_vizualization**](./gtdb_vizualization) </b> - Bacteria and Archae taxonomy <br>
- <b> [**gunc_visualization**](./gunc_visualization) </b> - GUNC output visualization of contig annotations <br>
- <b> [**ShortReads_from_LongReads**](./ShortReads_from_LongReads) </b> - converting long Pacbio reads into short PE Illumina reads <br>

![alt text](we-are2_small2.PNG)