{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Script is designed to create the taxonomy table file with NCBI tax-id\n",
    "\n",
    "Input is: <br>\n",
    "\n",
    "1. <i> File \"Tax_gtdb-ncbi.csv\" <br></i>\n",
    "Taxonomy table file. The table includes the gtdb taxonomy and ncbi names for all species for which I find a correspondence between dtdb and ncbi spesies names. \n",
    "Gtdb species names vs ncbi species names match files were taken from here: <br>\n",
    "https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files/\n",
    "\n",
    "2. <i> File \"names.dmp\" <br></i>\n",
    "File with ncbi name and ncbi tax-id. From zip folder: ncbi_taxdump_20220917 <br>\n",
    "https://data.gtdb.ecogenomic.org/releases/latest/auxillary_files/ <br>\n",
    "<br>\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# import packages\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Assign NCBI tax_id to GTDB species"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tax_id</th>\n",
       "      <th>name_txt</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>root</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2</td>\n",
       "      <td>Bacteria</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6</td>\n",
       "      <td>Azorhizobium</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>7</td>\n",
       "      <td>Azorhizobium caulinodans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>9</td>\n",
       "      <td>Buchnera aphidicola</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   tax_id                  name_txt\n",
       "0       1                      root\n",
       "1       2                  Bacteria\n",
       "2       6              Azorhizobium\n",
       "3       7  Azorhizobium caulinodans\n",
       "4       9       Buchnera aphidicola"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# read file with GTDB and NCBI taxonomies mach from \"NCBI_vs_GTDB\" scripts folder\n",
    "rel='v220'\n",
    "df1 = pd.read_excel('/Users/katerynapantiukh/Documents/1MyDisk/Code/Microbiome/NCBI_vs_GTDB/results/Tax_gtdb-ncbi_'+str(rel)+'.xlsx')\n",
    "# read \"names.dmp\" file from NCBI taxonomy\n",
    "name = pd.read_csv('/Users/katerynapantiukh/Documents/1MyDisk/Code/Microbiome/NCBI_tax_id/input/names.dmp', sep = '\\t', header = None)\n",
    "name.columns = ['tax_id','1', 'name_txt', '3','unique_name','5','name_class','7']\n",
    "name = name[name['name_class'] == 'scientific name']\n",
    "name = name.drop(columns = ['name_class', 'unique_name','1', '3', '5', '7'])\n",
    "df2 = name.reset_index(drop = True)\n",
    "df2.head(5)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>sp_gtdb</th>\n",
       "      <th>sp_ncbi</th>\n",
       "      <th>tax_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>Methanobacteriota</td>\n",
       "      <td>Methanobacteria</td>\n",
       "      <td>Methanobacteriales</td>\n",
       "      <td>Methanobacteriaceae</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>Methanobrevibacter sp017432245</td>\n",
       "      <td>Methanobrevibacter sp017432245</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>Methanobacteriota</td>\n",
       "      <td>Methanobacteria</td>\n",
       "      <td>Methanobacteriales</td>\n",
       "      <td>Methanobacteriaceae</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>Methanobrevibacter sp017410345</td>\n",
       "      <td>Methanobrevibacter sp017410345</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>Methanobacteriota</td>\n",
       "      <td>Methanobacteria</td>\n",
       "      <td>Methanobacteriales</td>\n",
       "      <td>Methanobacteriaceae</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>Methanobrevibacter sp017646885</td>\n",
       "      <td>Methanobrevibacter sp017646885</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>Methanobacteriota</td>\n",
       "      <td>Methanobacteria</td>\n",
       "      <td>Methanobacteriales</td>\n",
       "      <td>Methanobacteriaceae</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>Methanobrevibacter sp024409185</td>\n",
       "      <td>Methanobrevibacter sp024409185</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Archaea</td>\n",
       "      <td>Methanobacteriota</td>\n",
       "      <td>Methanobacteria</td>\n",
       "      <td>Methanobacteriales</td>\n",
       "      <td>Methanobacteriaceae</td>\n",
       "      <td>Methanobrevibacter</td>\n",
       "      <td>Methanobrevibacter sp900314635</td>\n",
       "      <td>Methanobrevibacter sp900314635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19690</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Pseudomonadota</td>\n",
       "      <td>Gammaproteobacteria</td>\n",
       "      <td>Pseudomonadales</td>\n",
       "      <td>Pseudomonadaceae</td>\n",
       "      <td>Stutzerimonas</td>\n",
       "      <td>Stutzerimonas zhaodongensis</td>\n",
       "      <td>Stutzerimonas zhaodongensis</td>\n",
       "      <td>[Pseudomonas] zhaodongensis</td>\n",
       "      <td>1176257.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19691</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Actinomycetota</td>\n",
       "      <td>Actinomycetes</td>\n",
       "      <td>Propionibacteriales</td>\n",
       "      <td>Propionibacteriaceae</td>\n",
       "      <td>Arachnia</td>\n",
       "      <td>Arachnia massiliensis</td>\n",
       "      <td>Arachnia massiliensis</td>\n",
       "      <td>[Pseudopropionibacterium] massiliense</td>\n",
       "      <td>2220000.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19692</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Lachnospirales</td>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>Ruminococcus_B</td>\n",
       "      <td>Ruminococcus_B gnavus</td>\n",
       "      <td>Ruminococcus_B gnavus</td>\n",
       "      <td>[Ruminococcus] gnavus 91.75%, (g__)</td>\n",
       "      <td>33038.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19693</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Lachnospirales</td>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>Mediterraneibacter</td>\n",
       "      <td>Mediterraneibacter lactaris</td>\n",
       "      <td>Mediterraneibacter lactaris</td>\n",
       "      <td>[Ruminococcus] lactaris 81.97%, (g__)</td>\n",
       "      <td>46228.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19694</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacillota_A</td>\n",
       "      <td>Clostridia</td>\n",
       "      <td>Lachnospirales</td>\n",
       "      <td>Lachnospiraceae</td>\n",
       "      <td>Mediterraneibacter</td>\n",
       "      <td>Mediterraneibacter torques</td>\n",
       "      <td>Mediterraneibacter torques</td>\n",
       "      <td>[Ruminococcus] torques 72.94%, (g__) 23.53%, F...</td>\n",
       "      <td>33039.0, 290055.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>19695 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         domain             phylum                class                order  \\\n",
       "0       Archaea  Methanobacteriota      Methanobacteria   Methanobacteriales   \n",
       "1       Archaea  Methanobacteriota      Methanobacteria   Methanobacteriales   \n",
       "2       Archaea  Methanobacteriota      Methanobacteria   Methanobacteriales   \n",
       "3       Archaea  Methanobacteriota      Methanobacteria   Methanobacteriales   \n",
       "4       Archaea  Methanobacteriota      Methanobacteria   Methanobacteriales   \n",
       "...         ...                ...                  ...                  ...   \n",
       "19690  Bacteria     Pseudomonadota  Gammaproteobacteria      Pseudomonadales   \n",
       "19691  Bacteria     Actinomycetota        Actinomycetes  Propionibacteriales   \n",
       "19692  Bacteria        Bacillota_A           Clostridia       Lachnospirales   \n",
       "19693  Bacteria        Bacillota_A           Clostridia       Lachnospirales   \n",
       "19694  Bacteria        Bacillota_A           Clostridia       Lachnospirales   \n",
       "\n",
       "                     family               genus  \\\n",
       "0       Methanobacteriaceae  Methanobrevibacter   \n",
       "1       Methanobacteriaceae  Methanobrevibacter   \n",
       "2       Methanobacteriaceae  Methanobrevibacter   \n",
       "3       Methanobacteriaceae  Methanobrevibacter   \n",
       "4       Methanobacteriaceae  Methanobrevibacter   \n",
       "...                     ...                 ...   \n",
       "19690      Pseudomonadaceae       Stutzerimonas   \n",
       "19691  Propionibacteriaceae            Arachnia   \n",
       "19692       Lachnospiraceae      Ruminococcus_B   \n",
       "19693       Lachnospiraceae  Mediterraneibacter   \n",
       "19694       Lachnospiraceae  Mediterraneibacter   \n",
       "\n",
       "                                   sp                         sp_gtdb  \\\n",
       "0      Methanobrevibacter sp017432245  Methanobrevibacter sp017432245   \n",
       "1      Methanobrevibacter sp017410345  Methanobrevibacter sp017410345   \n",
       "2      Methanobrevibacter sp017646885  Methanobrevibacter sp017646885   \n",
       "3      Methanobrevibacter sp024409185  Methanobrevibacter sp024409185   \n",
       "4      Methanobrevibacter sp900314635  Methanobrevibacter sp900314635   \n",
       "...                               ...                             ...   \n",
       "19690     Stutzerimonas zhaodongensis     Stutzerimonas zhaodongensis   \n",
       "19691           Arachnia massiliensis           Arachnia massiliensis   \n",
       "19692           Ruminococcus_B gnavus           Ruminococcus_B gnavus   \n",
       "19693     Mediterraneibacter lactaris     Mediterraneibacter lactaris   \n",
       "19694      Mediterraneibacter torques      Mediterraneibacter torques   \n",
       "\n",
       "                                                 sp_ncbi             tax_id  \n",
       "0                                                    NaN                NaN  \n",
       "1                                                    NaN                NaN  \n",
       "2                                                    NaN                NaN  \n",
       "3                                                    NaN                NaN  \n",
       "4                                                    NaN                NaN  \n",
       "...                                                  ...                ...  \n",
       "19690                        [Pseudomonas] zhaodongensis          1176257.0  \n",
       "19691              [Pseudopropionibacterium] massiliense          2220000.0  \n",
       "19692                [Ruminococcus] gnavus 91.75%, (g__)            33038.0  \n",
       "19693              [Ruminococcus] lactaris 81.97%, (g__)            46228.0  \n",
       "19694  [Ruminococcus] torques 72.94%, (g__) 23.53%, F...  33039.0, 290055.0  \n",
       "\n",
       "[19695 rows x 10 columns]"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# add tax-id to the table\n",
    "# Function to extract all species names from sp_ncbi\n",
    "def extract_species_names(sp_ncbi):\n",
    "    if pd.isna(sp_ncbi):\n",
    "        return []\n",
    "    species_names = []\n",
    "    for name in sp_ncbi.split(','):\n",
    "        parts = name.strip().split()\n",
    "        if len(parts) >= 2:\n",
    "            species_names.append(parts[0] + \" \" + parts[1])\n",
    "    return species_names\n",
    "\n",
    "# Expand sp_ncbi to multiple rows for each species name\n",
    "expanded_rows = []\n",
    "for idx, row in df1.iterrows():\n",
    "    species_names = extract_species_names(row['sp_ncbi'])\n",
    "    for name in species_names:\n",
    "        expanded_rows.append([row['sp_gtdb'], name])\n",
    "\n",
    "expanded_df = pd.DataFrame(expanded_rows, columns=['sp_gtdb', 'species_name'])\n",
    "\n",
    "# Merge expanded_df with df2\n",
    "merged_df = pd.merge(expanded_df, df2, how='left', left_on='species_name', right_on='name_txt')\n",
    "\n",
    "# Group by sp_gtdb and aggregate tax_ids\n",
    "grouped_df = merged_df.groupby('sp_gtdb')['tax_id'].apply(lambda x: ', '.join(x.dropna().astype(str))).reset_index()\n",
    "\n",
    "# Merge the aggregated tax_ids back to the original df1\n",
    "final_df = pd.merge(df1, grouped_df, how='left', on='sp_gtdb')\n",
    "\n",
    "final_df.to_excel('results/TaxID_gtdb-ncbi_'+str(rel)+'.xlsx', index=False)\n",
    "final_df\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Random check merge table"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>domain</th>\n",
       "      <th>phylum</th>\n",
       "      <th>class</th>\n",
       "      <th>order</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>sp</th>\n",
       "      <th>sp_gtdb</th>\n",
       "      <th>sp_ncbi</th>\n",
       "      <th>tax_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>15723</th>\n",
       "      <td>Bacteria</td>\n",
       "      <td>Bacteroidota</td>\n",
       "      <td>Bacteroidia</td>\n",
       "      <td>Bacteroidales</td>\n",
       "      <td>Marinifilaceae</td>\n",
       "      <td>Odoribacter</td>\n",
       "      <td>Odoribacter splanchnicus</td>\n",
       "      <td>Odoribacter splanchnicus</td>\n",
       "      <td>Odoribacter splanchnicus 89.15%, (g__)</td>\n",
       "      <td>28118.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         domain        phylum        class          order          family  \\\n",
       "15723  Bacteria  Bacteroidota  Bacteroidia  Bacteroidales  Marinifilaceae   \n",
       "\n",
       "             genus                        sp                   sp_gtdb  \\\n",
       "15723  Odoribacter  Odoribacter splanchnicus  Odoribacter splanchnicus   \n",
       "\n",
       "                                      sp_ncbi   tax_id  \n",
       "15723  Odoribacter splanchnicus 89.15%, (g__)  28118.0  "
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "final_df.loc[final_df['sp_gtdb'] == 'Odoribacter splanchnicus']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "For db version v214\n",
      "Total GTDB species in initial table is: 12214\n",
      "Total NCBI species in initial table  is: 2442982\n",
      "Total species in GTDB-NCBI result table is  12214\n"
     ]
    }
   ],
   "source": [
    "print('For db version',rel)\n",
    "print('Total GTDB species in initial table is:',len(df1))\n",
    "print('Total NCBI species in initial table  is:',len(df2))\n",
    "print('Total species in GTDB-NCBI result table is ',len(final_df))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "For db version v220\n",
      "Total GTDB species in initial table is: 19695\n",
      "Total NCBI species in initial table  is: 2442982\n",
      "Total species in GTDB-NCBI result table is  19695\n"
     ]
    }
   ],
   "source": [
    "print('For db version',rel)\n",
    "print('Total GTDB species in initial table is:',len(df1))\n",
    "print('Total NCBI species in initial table  is:',len(df2))\n",
    "print('Total species in GTDB-NCBI result table is ',len(final_df))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
